drivers/peci/peci_mchp_xec.c

```diff
@@ -170,16 +170,18 @@ static int peci_xec_write(struct device *dev, struct peci_msg *msg)
 	k_busy_wait(PECI_IO_DELAY);
 
 	/* Wait for transmission to complete */
+	uint8_t timeout_cnt = tx_buf->len + 1;
 #ifdef CONFIG_PECI_INTERRUPT_DRIVEN
-	if (k_sem_take(&peci_data.tx_lock, PECI_IO_DELAY * tx_buf->len)) {
+	if (k_sem_take(&peci_data.tx_lock, PECI_IO_DELAY * timeout_cnt)) {
 		return -ETIMEDOUT;
 	}
 #else
-	wait_timeout = tx_buf->len;
+	wait_timeout = timeout_cnt;
 	while (!(base->STATUS1 & MCHP_PECI_STS1_EOF)) {
 		k_busy_wait(PECI_IO_DELAY);
 
-		if (!wait_timeout) {
```

Comment:
with current code(master), wait_timeout never decrements. hence the loop is waiting infinity for  (!(base->STATUS1 & MCHP_PECI_STS1_EOF)) to becode true.
With this fix, the timeout is bounded hence i observed that most of the commands are failing due to insufficient wait time.
Is this fix tested for ping and other peci commands?

Reply:
@VenkatKotakonda Thanks for checking out this, will close this until I have HW access to debug further since it seems this is not a timeout problem.